pkg/controller/pxcrestore/controller.go

```diff
@@ -102,305 +105,317 @@ func (r *ReconcilePerconaXtraDBClusterRestore) Reconcile(ctx context.Context, re
 	}
 
-	log.Info("starting cluster", "cluster", cr.Spec.PXCCluster)
-	err = r.setStatus(cr, api.RestoreStartCluster, "")
+	restorer, err := r.getRestorer(ctx, cr, bcp, cluster)
 	if err != nil {
-		err = errors.Wrap(err, "set status")
-		return rr, err
+		return rr, errors.Wrap(err, "failed to get restorer")
 	}
 
-	if cr.Spec.PITR != nil {
-		oldSize := cluster.Spec.PXC.Size
-		oldUnsafePXCSize := cluster.Spec.Unsafe.PXCSize
-		oldUnsafeProxySize := cluster.Spec.Unsafe.ProxySize
-
-		var oldProxySQLSize int32
-		if cluster.Spec.ProxySQL != nil {
-			oldProxySQLSize = cluster.Spec.ProxySQL.Size
-		}
-		var oldHAProxySize int32
-		if cluster.Spec.HAProxy != nil {
-			oldHAProxySize = cluster.Spec.HAProxy.Size
+	switch statusState {
+	case api.RestoreNew:
+		if cr.Spec.PITR != nil {
+			err = backup.CheckPITRErrors(ctx, r.client, r.clientcmd, cluster)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+
+			annotations := cr.GetAnnotations()
+			_, unsafePITR := annotations[api.AnnotationUnsafePITR]
+			cond := meta.FindStatusCondition(bcp.Status.Conditions, api.BackupConditionPITRReady)
+			if cond != nil && cond.Status == metav1.ConditionFalse && !unsafePITR {
+				statusMsg = fmt.Sprintf("Backup doesn't guarantee consistent recovery with PITR. Annotate PerconaXtraDBClusterRestore with %s to force it.", api.AnnotationUnsafePITR)
+				statusState = api.RestoreFailed
+				return reconcile.Result{}, nil
+			}
 		}
 
-		cluster.Spec.Unsafe.PXCSize = true
-		cluster.Spec.Unsafe.ProxySize = true
-		cluster.Spec.PXC.Size = 1
-
+		err = validate(ctx, restorer, cr)
+		if err != nil {
+			if errors.Is(err, errWaitValidate) {
+				return rr, nil
+			}
+			statusMsg = fmt.Sprintf("failed to validate restore job: %s", err.Error())
+			statusState = api.RestoreFailed
+			return rr, err
+		}
+		cr.Status.PXCSize = cluster.Spec.PXC.Size
 		if cluster.Spec.ProxySQL != nil {
-			cluster.Spec.ProxySQL.Size = 0
+			cr.Status.ProxySQLSize = cluster.Spec.ProxySQL.Size
 		}
 		if cluster.Spec.HAProxy != nil {
-			cluster.Spec.HAProxy.Size = 0
+			cr.Status.HAProxySize = cluster.Spec.HAProxy.Size
 		}
-
-		if err := k8s.UnpauseClusterWithWait(ctx, r.client, cluster); err != nil {
-			return rr, errors.Wrap(err, "restart cluster for pitr")
+		cr.Status.Unsafe = cluster.Spec.Unsafe
+
+		log.Info("stopping cluster", "cluster", cr.Spec.PXCCluster)
+		statusState = api.RestoreStopCluster
+	case api.RestoreStopCluster:
+		// TODO: we should use PauseCluster and delete PVCs
+		{
+			err := k8s.PauseClusterWithWait(ctx, r.client, cluster, true)
+			if err != nil {
+				return rr, errors.Wrapf(err, "stop cluster %s", cluster.Name)
+			}
+			/*
+				paused, err := k8s.PauseCluster(ctx, r.client, cluster)
+				if err != nil {
+					return rr, errors.Wrapf(err, "stop cluster %s", cluster.Name)
+				}
+				if !paused {
+					log.Info("waiting for cluster to stop", "cluster", cr.Spec.PXCCluster, "msg", err.Error())
+					return rr, nil
+				}
+			*/
```

Comment:
do we need this commented code?

Reply:
https://github.com/percona/percona-xtradb-cluster-operator/pull/1610/commits/f8b07591d3e50248e70175a08f9b25356ea3f93d